sklearn/linear_model/tests/test_base.py

```diff
@@ -206,6 +220,61 @@ def test_linear_regression_sparse_multiple_outcome(random_state=0):
     assert_array_almost_equal(np.vstack((y_pred, y_pred)).T, Y_pred, decimal=3)
 
 
+def test_linear_regression_positive():
+    # Test nonnegative LinearRegression on a simple dataset.
+    # a simple dataset
+    X = [[1], [2]]
+    Y = [1, 2]
+
+    reg = LinearRegression(positive=True)
+    reg.fit(X, Y)
+
+    assert_array_almost_equal(reg.coef_, [1])
+    assert_array_almost_equal(reg.intercept_, [0])
+    assert_array_almost_equal(reg.predict(X), [1, 2])
+
+    # test it also for degenerate input
+    X = [[1]]
+    Y = [0]
+
+    reg = LinearRegression(positive=True)
+    reg.fit(X, Y)
+    assert_allclose(reg.coef_, [0])
+    assert_allclose(reg.intercept_, [0])
+    assert_allclose(reg.predict(X), [0])
+
+
+def test_linear_regression_positive_multiple_outcome(random_state=0):
+    # Test multiple-outcome nonnegative linear regressions
+    random_state = check_random_state(random_state)
+    X, y = make_sparse_uncorrelated(random_state=random_state)
+    Y = np.vstack((y, y)).T
+    n_features = X.shape[1]
+
+    ols = LinearRegression(positive=True)
+    ols.fit(X, Y)
+    assert ols.coef_.shape == (2, n_features)
+    assert np.all(ols.coef_ >= 0.)
+    Y_pred = ols.predict(X)
+    ols.fit(X, y.ravel())
+    y_pred = ols.predict(X)
+    assert_allclose(np.vstack((y_pred, y_pred)).T, Y_pred)
+
+
+def test_linear_regression_positive_vs_nonpositive():
+    # Test nonnegative LinearRegression on a simple dataset.
+    # a simple dataset
+    X = [[1], [2]]
+    Y = [1, 2]
+
+    reg = LinearRegression(positive=True)
+    reg.fit(X, Y)
+    regn = LinearRegression(positive=False)
+    regn.fit(X, Y)
+
+    assert not np.array_equal(reg.coef_, regn.coef_)
```

Comment:
we should instead check that there is a significant difference between the coeffs, because it is too easy to fail a `np.array_equal` check due to numerical instability.

Something like `assert np.mean((reg - regn)**2) > 1e-3` to check the MSE